### DISCUSSION TITLE
How can I `useFetcher` to retrieve binary data?

### DISCUSSION THREAD

**Discussion by connor4312** [remix-run/remix#2618] [Q&A]
I have a large data structure that one of my API routes is packing into a binary response. On that response, in spite of setting its `Content-Type` to `application/octet-stream`, it gets passed as a string in the fetcher `data`.

![image](https://user-images.githubusercontent.com/2230985/161444703-4dcf46c1-8217-42f9-afeb-45d2eed13d2f.png)

Is there a way to get around this and access it as an arrayBuffer or uint8array?

**Comment by kiliman** [comment]
Just like loaders and actions, `useFetcher` expects responses to be JSON format.

If you need non-JSON data, you will need to use the standard `fetch` function.

It would be nice if Remix honored the `content-type`.

**Reply by sergiodxa** [reply]
I was thinking that too, it would be useful if useFetcher at least could check the Content-Type and decide how to read the response based on that, use `response.json()` only for JSON response and other formats for the rest, and maybe use `response.blob()` or `response.arrayBuffer()` from other content types.

**Reply by connor4312** [reply]
Definitely agreed, content-type matching would be perfect.

**Comment by R3al0n** [comment]
Is there any update on this? or is there any code example available for binary data use case?

**Reply by blissito** [reply]
I use fetch. And, there is the [
unstable_parseMultipartFormData](https://remix.run/docs/en/main/utils/parse-multipart-form-data) too, which gives you the `arrayBuffer `
